src/util.coffee

```diff
@@ -32,3 +32,25 @@
                                       node?.tag is 'tag:yaml.org,2002:map'
 
 @NON_PRINTABLE = /[^\x09\x0A\x0D\x20-\x7E\x85\xA0-\uD7FF\uE000-\uFFFD]/
+
+@stringifyJSONCircularDepth = (json, depth = 1000) ->
```

Comment:
@derrickburns @dmartinezg If you want to abort circulars dependencies by default. It's not a problem, but you should allow the possibility to parse circular schemas with dereference and a max depth.

Is this aligned with your thoughts ?